### PR TITLE
Clean addition of DIP0001

### PR DESCRIPTION
**Pull request by DarrenTapp**
This is a very clean addition of DIP0001

### CODE REVIEW

**Comment by andyfreer** [comment]
ACK

File: dip-0001.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        }
+```
+
+Otherwise, the next state depends on the previous state:
+
+```C++
+switch (GetStateForBlock(GetAncestorAtHeight(block, block.height - 4032))) {
+```
+
+We remain in the DEFINED state until either we pass the starttime or the timeout. GetMedianTimePast in the code below refers to the median nTime of a block and its 10 predecessors.
```
Comment (willwray) [inline dip-0001.md]: We remain in the DEFINED state until we either pass the starttime or the timeout.

File: dip-0001.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+## Selection of Parameters
+
+For values of starttime and timeout, we suggest:
+* starttime= 1508025600--turn of Oct 15 at midnight 
+* timeout= 1539561600-- 52 weeks later
+
+This is assuming the code is available mid September.  If the code is available at a later release these numbers should be adjusted.
+
+The selection of a round consisting of 4032 blocks is made based on masternodes not being randomly selected, rather they are in a queue.  As such it can’t treat the sample of masternodes as a simple random sample.  At current time 4032 masternodes would consist of a sample of around 87% of masternodes.  This number is small enough that the sample of masternodes are unique. If this number were larger there would be a chance that some masternodes would be sampled twice while others aren’t sampled at all.  This property is maintained as long as the number of blocks in a round is under 90% of all masternodes.  The number 4032 also has the property of being equal to about a week of blocks.
```
Comment (willwray) [inline dip-0001.md]: If this number were larger then

File: dip-0001.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+The selection of a round consisting of 4032 blocks is made based on masternodes not being randomly selected, rather they are in a queue.  As such it can’t treat the sample of masternodes as a simple random sample.  At current time 4032 masternodes would consist of a sample of around 87% of masternodes.  This number is small enough that the sample of masternodes are unique. If this number were larger there would be a chance that some masternodes would be sampled twice while others aren’t sampled at all.  This property is maintained as long as the number of blocks in a round is under 90% of all masternodes.  The number 4032 also has the property of being equal to about a week of blocks.  
+
+## Calculation of thresholdreached
+
+### Determination of thresholdreached
+
+Given a block with block.height % 4032 = 0 we can calculate thresholdreached.  Thresholdreached is determined by bit1 on each version of all blocks in the previous round.  Note the current block is not considered to be in the previous round.  We define thresholdreached to be true when and only when,
```
Comment (willwray) [inline dip-0001.md]: (format code and thresholdreached with back-ticks)

Given a block with `block.height % 4032 = 0` we can calculate `thresholdreached`.  `thresholdreached` is determined by bit1 on each version of all blocks in the previous round.  Note the current block is not considered to be in the previous round.  We define `thresholdreached` to be true when and only when:

Comment (DarrenTapp) [inline dip-0001.md]: patch 3 should address all of the comments.  I hope I did that correctly.

**Pull request closed without merging**